### ISSUE TITLE
Facility Locator Map crashes on Chrome

### ISSUE THREAD

**Issue by eliotwinchell** [department-of-veterans-affairs/vets-website#9165]
Title: Facility Locator Map crashes on Chrome

Comment: When a user goes to the Find Facility page, the map/page will crash if the user pans around the map. It will crash if they pan the map after entering a location, or it will crash if they try to pan the map around after page load. Many latitudes/longitudes are being asynchronously loaded, and the page fails to recover. There are errors from raven.js in the console stating: "Throttling history state changes to prevent the browser from hanging." There are also issues when the user attempts to zoom in/out. The view is refreshing to the previous zoom level, so the user is unable to zoom in/out. The user panning the map on Safari did not crash the page and worked as expected, this bug only appeared on Google Chrome. The zooming in/out bug appeared in both browsers. 

The environment you experienced the bug: Google Chrome 70.0.3538.110 (Official Build) (64-bit)

The exact steps you took that triggered the bug: Went to https://www.va.gov/find-locations/, entered a location, and panned around the map. 

The expected outcome: A refresh of VA Facilities of the new user map view.  

The actual outcome, including screen shot: Map/page crash. screen recording: https://streamable.com/ondsu

I would like to fix this bug in the facility-locator application, and submit a PR, if it's not being fixed already.

**Comment by ncksllvn** [comment]
Hi @eliotwinchell! Thank you for logging this issue and the level of detail in doing so. I especially appreciate the Streamable video. My team has this issue scheduled to be completed during our current sprint, but we're also curious if you have any interest in lending a hand and contributing to the project with the work required here? No pressure, but we appreciate any help the community is willing to offer!

**Comment by eliotwinchell** [comment]
Hi @ncksllvn 
I'm happy to provide any help I can! I looked at the VAMap.jsx this weekend, ran a local build of this repo, and did some experimenting. I found that the "this.props.updateSearchQuery(.....)" block in the handleBoundsChanged may be the issue. There's a position property (current lat/lng) being updated in the search query, and when that position property is removed, the panning no longer crashes. Is that position property needed when re-searching the bounds?? I think only the bounds are being used to search for facilities. It also makes the zooming smoother. Will submit a PR once I can confirm that it's not needed. This file needs some refactoring though, I've seen @r-mc2 notes and would like to help out with that

**Comment by r-mc2** [comment]
The position should only be needed once all panning/zooming has completed. The intermediary coords as the map is moving are easily ignorable.

For reference, that center point is saved to the Redux store and referenced later on for searches via `genBBoxFromCoords` so the appropriate bounding box is sent to the API. It is also referenced in `updateUrlParams` to update the URL in the address bar to ensure it is a shareable link for users to send to others.

I know the function you pointed out, `handleBoundsChange`, is registered as a listener to the MapBox map we are using to handle whenever the map sends out an event of the bounds having changed (i.e. the user panning).
```jsx
<Map ref="map" center={position} zoomSnap={0.5} zoomDelta={0.5}
  zoom={parseInt(currentQuery.zoomLevel, 10)} style={{ minHeight: '75vh', width: '100%' }}
  scrollWheelZoom={false} onMoveEnd={this.handleBoundsChanged}>
```
Not sure if there is a way to throttle the incoming events to the handler though. Or even just say "let me know where you are when all movement is done." But I do not know if that is possible.
e.g. How do you know it is done? What if the user pans the map a little, then moves it a little more a few seconds later?
(Unfortunately I did not have any hand in the map itself being integrated into the component so my knowledge of that library is lacking)

**Comment by ncksllvn** [comment]
Thank you for your help so far @eliotwinchell!

**Comment by bshyong** [comment]
In case this helps: The app is using React Leaflet for the Map component, and there is currently a onMoveEnd trigger that calls handleBoundsChanged 


https://leafletjs.com/reference-1.3.4.html#map-moveend

**Comment by ncksllvn** [comment]
A patch for this issue was implemented here, https://github.com/department-of-veterans-affairs/vets-website/pull/9259, and will be live at roughly 3 PM ET. Thank you @eliotwinchell for raising this and for your effort in investigating a fix. Truly appreciated!

Thanks @r-mc2 and @bshyong for your insight!

**Comment by eliotwinchell** [comment]
Hey @ncksllvn,
I added the code change in the #9259 PR, and it did not fix the issue I think, although that code change does make that JSX variable declaration cleaner! I think the bug lies in the position prop being updated in updateSearchQuery in the handleBoundsChanged function. The Mapbox center is using the position prop, and it seems like the current method (as @r-mc2 and @bshyong) pointed out of updating it with handleBoundsChanged is creating an infinite loop. I have a bit of a lack of knowledge in Redux so I'm going through the docs and will try to provide more help! 

I've included a streamable link, which shows the new facilityLocatorMarkers declaration/conditional, I cleared my cache before recording as well. I put in a console.log in the handleBoundsChanged function and have the console open demonstrating the infinite loop

https://streamable.com/mqllc

**Comment by ncksllvn** [comment]
Thanks for following up @eliotwinchell! 

Hm 🤔That's interesting, because I'm unable to replicate it anymore. That PR entailed a dependency update - did you run a `yarn install` prior to rebuilding the site?

**Comment by eliotwinchell** [comment]
Ahh @ncksllvn,
thank you, working now. I definitely was not going down the right path, was that error about the facility marker appearing in the console? Thanks for fixing that, I will see if there's more bugs I can try to fix, learning a lot!

**Comment by ncksllvn** [comment]
Of course @eliotwinchell, thank you! It was actually just an internal issue with the Map component provided by the `react-leaflet` package. I observed this by just a `console.log` each time `onMoveEnd` was called by the module. Upgrading that package fixed the issue, but then I did see an error in the console about the `FeatureGroup`, so I had to patch that issue afterwards.

Truly appreciate your help!
